src/test/java/com/twilio/Assert.java

```diff
@@ -30,15 +30,18 @@ public static void assertUrlsEqual(final String expected, final URL actual) {
     }
 
     public static void assertUrlsEqual(final String expected, final String actual) {
-        String[] expectedParts = expected.split("\\?");
-        String[] actualParts = actual.split("\\?");
+        String[] expectedParts = expected.split("[?#]");
+        String[] actualParts = actual.split("[?#]");
 
         assertEquals(expectedParts.length, actualParts.length);
         assertEquals(expectedParts[0], actualParts[0]);
 
-        if (expectedParts.length == 2) {
+        if (expectedParts.length >= 2) {
             assertQueryStringsEqual(expectedParts[1], actualParts[1]);
         }
+        if (expectedParts.length == 3) {
```

Comment:
Why is this case called out separately, when the above code on line 39 checks for all values `>=2`? Should we call both functions if `expectedParts.length == 3`?

Reply:
Note the code inside the conditional blocks. If the number of parts is >= 2, then the url contains a query string, which are there compared. If number of parts is 3, then the url also contains a fragment, which then needs to be compared as well.